File: backend/models/room.go

package models

type Room struct {
  RoomId string `bson:"_id,omitempty" json:"id"`;
  OwnerId string `bson:"ownerId" json:"ownerId"`;
  MemeberIds []RoomUser `bson:"memberIds" json:"memberIds"`;
}

type RoomUser struct{
  UserId string `bson:"userId" json:"userId"`;
  X int `bson:"x" json:"x"`;
  Y int `bson:"y" json:"y"`;
  Direction string `bson:"direction" json:"direction"`;
}
// direction would follow the clockwise numbering starting with 0 to top

File: backendc#/virutalOffice/Domain/Room.cs

public class Room{
  public int RoomId {get;set;}
  public int OwnerId {get;set;}
  public List<UserPosition>UserIds {get;set;} = [];
}

public class UserPosition{
  public int UserId {get;set;}
  public int PosX {get;set;}
  public int PosY {get;set;}
}
